### ISSUE TITLE
StringLike field with quotes breaks out of title

### ISSUE THREAD

**Issue by DRSisco** [railsadminteam/rails_admin#3686]
**Describe the bug**

Storing content in a field that contains quotes results in only the content before the first quote is in the title. The rest of the content is stored in the tag as individual attributes. 

_We noticed this when the word `hidden` was stored after a quote, which resulted in the cell being hidden and the rest of the row shifting to the left._

**Reproduction steps**
- For a given record with a StringLike field:
  - Put the content `Example "text" hidden ` in the field.
  - View the index page that contains this model/field. 

**Expected behavior**
- The content should show (with ellipsis if necessary based on space), and the title should represent the whole field escaping quotes with `&quot;`

**Additional context**

- `rails` version: 7.1.3
- `rails_admin` version: 3.0.0.beta2

**Comment by mshibuya** [comment]
I realized that this issue lead to an XSS vulnerability, so I've published a security advisory: https://github.com/railsadminteam/rails_admin/security/advisories/GHSA-8qgm-g2vv-vwvc
I recommend upgrading to the latest release 3.1.3, as soon as possible. Also fixed in the master: https://github.com/railsadminteam/rails_admin/commit/32f91e4b49205e44d3931c2e36d9f7273384a250

Thank you for giving the valuable report 👍

**Comment by q3aiml** [comment]
@mshibuya thank you for your work on this!

When trying 3.1.3 with this change I am seeing unescaped HTML tags in the title attribute. The field value does appear properly escaped -- it is the HTML from pretty_value in rails admin that is unescaped and no longer stripped out.

For example on booleans:

![image](https://github.com/railsadminteam/rails_admin/assets/1438495/dbcf8485-aa8f-4cc4-950f-ec7b7162460f)

```
<td class="flag1_field boolean_type" title="<span class="badge bg-danger"><span class="fas fa-times"></span></span>">
  <span class="badge bg-danger"><span class="fas fa-times"></span></span>
</td>
```

Or associations:

![image](https://github.com/railsadminteam/rails_admin/assets/1438495/948bfd61-2f00-4424-b183-e134696060ae)

```
<td class="user_field belongs_to_association_type" title="<a href="/admin/broker/00000000-0000-0000-0000-000000000000">Test User1</a>">
  <a href="/admin/broker/00000000-0000-0000-0000-000000000000">Test User1</a>
</td>
```

- `rails` version: 7.0.8
- `rails_admin` version: 3.1.3

**Comment by mshibuya** [comment]
@q3aiml So sorry for the issue, I messed up 🙇‍♂️ 
I'm preparing for another release now.

**Comment by mshibuya** [comment]
Released 3.1.4. Please give it a try, and if you see any issue please let me know 🙇‍♂️
